rrmngmnt/nmcli.py

```diff
@@ -0,0 +1,644 @@
+                    - "name"
+                    - "uuid"
+                    - "type"
+                    - "device"
+
+        Raises:
+            CommandExecutionFailure: if the remote host returned a code
+            indicating a failure in execution.
+        """
+        cons = list()
+
+        out = self._exec_command(command="nmcli -t con show")
+        for line in out.splitlines():
+            properties = line.split(":")
+            cons.append(
+                {
+                    "name": properties[0],
+                    "uuid": properties[1],
+                    "type": properties[2],
+                    "device": properties[3],
+                }
+            )
+
+        return cons
+
+    def get_all_devices(self):
+        """
+        Gets existing devices details.
+
+        Returns:
+            list[dict]: each dict in the returned list represents a device,
+                and has the following keys:
+                    - "name"
+                    - "type"
+                    - "mac"
+                    - "mtu"
+        """
+        devices = list()
+        con_names = [con.get("name") for con in self.get_all_connections()]
+
+        for name in con_names:
+            out = self._exec_command(
+                command=(
+                    "nmcli -e no "
+                    "-g GENERAL.DEVICE,GENERAL.TYPE,GENERAL.HWADDR,GENERAL.MTU "  # noqa: E501
+                    "dev show {con}"
+                ).format(con=name)
+            )
+            properties = out.splitlines()
+            devices.append(
+                {
+                    "name": properties[0],
+                    "type": properties[1],
+                    "mac": properties[2],
+                    "mtu": properties[3],
+                }
+            )
+
+        return devices
+
+    def get_device_type(self, device):
+        """
+        Gets the device type.
+
+        Args:
+            device (str): device name.
+
+        Returns:
+            str: the device type.
+
+        Raises:
+            CommandExecutionFailure: if the remote host returned a code
+            indicating a failure in execution.
+        """
+        return self._exec_command(
+            command="nmcli -g GENERAL.TYPE device show {dev}".format(dev=device)  # noqa: E501
+        ).strip()
+
+    def set_connection_state(self, connection, state):
+        """
+        Sets a connection's state.
+
+        Args:
+            connection (str): name, UUID or path.
+            state (str): the desired state.
+                available states are: ["up", "down"].
+
+        Raises:
+            CommandExecutionFailure: if the remote host returned a code
+            indicating a failure in execution.
+        """
+        self._exec_command(
+            command="nmcli connection {state} {con}".format(state=state, con=connection)  # noqa: E501
+        )
+
+    def add_ethernet_connection(
+        self,
+        con_name,
```

Comment:
'con_name' could be 'name' since the function is called 'add_ethernet_connection'